state.c

```diff
@@ -429,24 +430,34 @@ int pv_state_start(struct pv_state *s)
 	return pv_state_mount_bsp_volumes(s);
 }
 
-static bool pv_state_can_start_platform(struct pv_state *s,
-					struct pv_platform *p)
+bool pv_state_check_goals(struct pv_state *s, struct pv_platform *p,
```

Comment:
My comment on this was based on my brain still not accepting that we were going to unblock groups after the timeout. But now I still don't love this. I'd say this functions  is now doing two things and thus could be split in two. The old pv_state_check_goals returns true if all goals from previous groups have been validated. However, what you are trying to put in here is more like pv_group_check_timeout? Maybe all the time logic could go into group.c? Not completely sure about these things, but for sure this stuff doesn't fit in pv_state_check_goals IMO